app/plugin.go

```diff
@@ -429,11 +431,29 @@ func (a *App) ShutDownPlugins() {
 	a.PluginEnv = nil
 }
 
+func getKeyHash(pluginId, key string) (string, *model.AppError) {
+	if len(pluginId) == 0 {
+		return "", model.NewAppError("getKeyHash", "app.plugin.key_value.plugin_id.app_error", nil, "", http.StatusBadRequest)
+	}
+
+	if len(key) == 0 {
+		return "", model.NewAppError("getKeyHash", "app.plugin.key_value.key.app_error", nil, "", http.StatusBadRequest)
+	}
+
+	hash := sha256.New()
+	hash.Write([]byte(pluginId + key))
```

Comment:
I think we should do something slightly different (pluginId = "ab", key = "c" is the same as pluginId = "a", key = "bc"). A better way to go would be to use an invalid character as a delimiter (null byte maybe?) or hash the id and key individually, then concatenate them.

But I'm actually wondering if it might be better to keep two separate plugin id and key columns. It seems like it'd be nice to be able to delete all data corresponding to a plugin when you delete it.

Reply:
Yeah I was actually thinking about that a bit while doing this, it's kind of nice to just be able to easily remove all the data for a single plugin with one easy query

I can just switch it back to two columns and hash both of them to dodge the. It's a little worse because we're doubling the work (hashing/encoding twice) but I think it's probably worth it

Reply:
Putting a limit on plugin ids and using plaintext for them would probably be fine too. Then only hash the keys. Plugin ids already have filesystem-imposed limits anyways since we use them in file paths.